Feature/components

lucasnetwork opened the pull request.

lucasnetwork committed:
feat(dependence and component): add styled-components dependence and add Button component

lucasnetwork committed:
feat(component): add Input component

lucasnetwork committed:
feat(component): add OptionBar component

create component OptionBar to use AsideBar component

lucasnetwork committed:
feat(button component): add style of Button when disabled

lucasnetwork committed:
feat(component): add header component

lucasnetwork committed:
feat(branch): merge develop branch

lucasnetwork committed:
feat(component): create Button component to use in bottom menu

lucasnetwork committed:
feat(component): add ProgramIcon component to use bottom open menu

lucasnetwork committed:
feat(component): create ExplicationProgramIcon to use in Bottom open menu

lucasnetwork committed:
feat(component): add menu component

lucasnetwork committed:
feat(component): add type color secondary in button and add aside responsive

lucasnetwork committed:
feat(service hook): add useDimension hook to verify width and height of component

lucasnetwork committed:
feat(component): create Main component

lucasnetwork committed:
feat(dependence): add resize-observer-polyfill dependence

lucasnetwork committed:
feat(component): add Modal component

lucasnetwork committed:
fix(component men): fix z-index of bottom menu

lucasnetwork committed:
feat(component): add select component

lucasnetwork committed:
feat(component): add optional properties

lucasnetwork committed:
feat(dependence): add react-select dependence

lucasnetwork committed:
feat(component): add Search component

lucasnetwork committed:
feat(component): add ListOption component

lucasnetwork committed:
feat(component): improvement Button in AsideBar when window is small

lucasnetwork committed:
feat(component): styling component select,adding icon dropdown and styling option selected

lucasnetwork committed:
feat(component): add component children in main component

lucasnetwork committed:
feat(component): create MultiSelect component

lucasnetwork committed:
feat(component): add labelStyling in input component

lucasnetwork committed:
feat(component): add htmlElement in props of ExplicationProgramIcon component

lucasnetwork committed:
feat(component): add animation in explicationProgramIconProps component when open menu

lucasnetwork committed:
feat(component): remove jsx of main component and change style.ts name to index.ts

lucasnetwork committed:
feat(component): add Content component

lucasnetwork committed:
feat(component): improvement props of ExplicationProgramIcon

lucasnetwork committed:
feat(component): create props of programs in Menu

lucasnetwork committed:
fix(component): fix labelStyle in Input component

change labelStyle to labelOptions and remove labelOptions of style prop to spread in label

lucasnetwork committed:
feat(component): fix desestructure of props open in styles of Modal

lucasnetwork committed:
feat(component): add style overflow in content component

lucasnetwork committed:
feat(component): create ButtonConfirm and HeaderButton

lucasnetwork committed:
feat(component): styling focus and placeholder of input component

lucasnetwork committed:
feat(component): add badge component

lucasnetwork committed:
feat(component): add children prop in asideBar

lucasnetwork committed:
feat(component): add card component

lucasnetwork committed:
feat(component): add onclick in prop progras in bottom menu

lucasnetwork committed:
feat(mock): add mocks

lucasnetwork committed:
feat(styling): add fontFamily ubuntu

lucasnetwork committed:
feat(tsconfig and html): add resolveJSONModule in tsconfig and add font link

lucasnetwork committed:
fix(component): fix hover of header button

lucasnetwork committed:
feat(component): add buttonLabel in header

lucasnetwork committed:
feat(component): add cursor pointer in buttons and add type button in headerButton

lucasnetwork committed:
feat(dependence): add svg-url-loader in webpack

lucasnetwork committed:
feat(component): fix image in explicationProgramIcon

lucasnetwork committed:
feat(component): improvement style in menu open

lucasnetwork committed:
feat(component): add text-overflow styling in program icon text

lucasnetwork committed:
feat(page): add initial page to test components

lucasnetwork committed:
feat(page): fix asidemenu contents

lucasnetwork committed:
feat(component): add props in Search component

lucasnetwork committed:
feat(component): add default background and colorText in Badge component

lucasnetwork committed:
feat(component): add props card in card component

lucasnetwork committed:
fix(component): fix padding of input component

lucasnetwork committed:
style(component): remove console.log

lucasnetwork committed:
feat(page): update page test

lucasnetwork committed:
feat(component): styling placeholder in multiSelect

lucasnetwork committed:
feat(page): update page test

filiperochalopes commented on core/ui/src/components/Badge/index.ts:
```diff
@@ -0,0 +1,21 @@
+import styled from 'styled-components'
```
@lucasnetwork A estrutura que sempre trabalhamos deve ser mantida, sempre com um arquivo `tsx` que renderiza html e outro arquivo `styles.ts` que são os estilos que serão utilizados daquele componente. Estou vendo em todos apenas um arquivo `index.ts(x)`

lucasnetwork replied:
resolvido

filiperochalopes commented on core/ui/src/components/Button/index.tsx:
```diff
@@ -0,0 +1,30 @@
+import styled from 'styled-components'
```
Todo arquivo `tsx` tem esse `x` no final porque renderiza `html` esse arquivo é um arquivo de estilos. deve seguir o nosso padrão `styles.ts` e ter um componente em `index.tsx` com esses estilos funcionando como `Wrap` e facilitando o uso do componente personalizado.

lucasnetwork replied:
resolvido

filiperochalopes commented on core/ui/src/components/Card/index.tsx:
```diff
@@ -0,0 +1,25 @@
+import Badge from '../Badge'
+import Container from './styles'
+
+interface CardProps{
```
Em `typescript` vamos padronizar que as Interfaces devem começar com `I`, por convenção. No caso seria `ICardProps`

lucasnetwork replied:
corrigido

filiperochalopes commented on core/ui/src/components/Card/index.tsx:
```diff
@@ -0,0 +1,25 @@
+interface CardProps{
+    subtitleTexts:{id:string, text:string}[];
+    title:string;
+    badges:{id:string;text:string;colorBackground?:string;colorText?:string}[]
```
Badge não é uma propriedade, pelo contrário, ele é um componente que pode ser renderizado no próprio título se necessário, e não algo a parte. Pense no badge como um badge do github, que colocamos no README no próprio texto.

lucasnetwork replied:
resolvido

filiperochalopes commented on core/ui/src/components/Card/index.tsx:
```diff
@@ -0,0 +1,25 @@
+const Card:React.FC<CardProps> = ({ subtitleTexts, badges, title }) => (
+    <Container>
+        <p className='title'>
+            {title} {badges.map(badge => <Badge key={badge.id} background={badge.colorBackground} colorText={badge.colorText}>
```
Esse title pode ser um componente `<span>Título bla bla bla <Badge color="green">Observação</Badge></span>` Mais simplicidade e menos regras.

lucasnetwork replied:
resolvido

filiperochalopes commented on core/ui/src/components/Form/MultiSelect/index.tsx:
```diff
@@ -0,0 +1,122 @@
+      options={[{ value: 'teste', label: 'teste' }]}
+      isMulti={true as IsMulti}
+      components={{
+        IndicatorsContainer: props => <div style={{ marginRight: '8px' }} {...props}><svg width="8" height="4" viewBox="0 0 8 4" fill="none" xmlns="http://www.w3.org/2000/svg" >
```
O que é um Indicator? Só lendo o código não consigo identificar, isso atribui uma baixa legibilidade do código.

lucasnetwork replied:
indicatorsContainer é a propria sintaxe do react-select para customizar os componentes dele.

filiperochalopes replied:
Ciente

filiperochalopes commented on core/ui/src/components/Form/Select/index.tsx:
```diff
@@ -0,0 +1,104 @@
+import React from 'react'
```
A organização das pastas deve ser diferente, lembra? padronizada, então seria `core/ui/src/Form/components/Select/index.tsx`

lucasnetwork replied:
na verdade foi dada essa estrutura:
- components
--- Window
---- components
----- Header
------ index.tsx
------ styles.ts
----- SideBar
------- index.tsx
------- styles.ts
--- SideBar
---- index.tsx
---- styles.ts

e que todos deveriam ficar dentro de componentes.

filiperochalopes replied:
Exatamente, vc acabou de repetir o que eu falei, observe como nessa estrutura de arquivos não existe `components/Window/Header`, mas sim `components/Window/components/Header`, sendo que esse Header deve ser usado exclusivamente pelo Window, se esse Header for usado em algum lugar mais ele deve estar em `components/Header`. Essa é uma estrutura que propõe escalabilidade e não é de minha cabeça, é o que sempre usamos baseado nesse artigo: https://alexmngn.medium.com/how-to-better-organize-your-react-applications-2fd3ea1920f1

lucasnetwork replied:
mas porque form fica fora da pasta components?

lucasnetwork replied:
resolvido

filiperochalopes commented on core/ui/src/components/Search/index.tsx:
```diff
@@ -0,0 +1,25 @@
+import React, { HTMLAttributes, InputHTMLAttributes } from 'react'
+import Input from '../Form/Input'
```
Importação relativa, não deve ocorrer. A não ser que esteja na mesma pasta ou pastas filho

lucasnetwork replied:
como já tinha dito antes, importação por paths relativos não esta funcionando

filiperochalopes replied:
Como eu comentei previamente, eles já foram corrigidos. Informei pelo WhatsApp que a atualização estava em `develop`

lucasnetwork replied:
corrigido

filiperochalopes commented on core/ui/src/components/Search/index.tsx:
```diff
@@ -0,0 +1,25 @@
+import React, { HTMLAttributes, InputHTMLAttributes } from 'react'
+import Input from '../Form/Input'
+import Container from './styles'
+/* global HTMLButtonElement */
```
O que são essas variáveis globais?

filiperochalopes commented on core/ui/src/assets/icons/loja.svg:
```diff
@@ -0,0 +1,184 @@
+<svg width="54" height="53" viewBox="0 0 54 53" fill="none" xmlns="http://www.w3.org/2000/svg">
```
O arquivo deve seguir um padrão de nomenclatura, se os outros estão em inglês, esse deve estar também `store.svg`

lucasnetwork replied:
resolvido

filiperochalopes commented on core/ui/src/components/Search/styles.ts:
```diff
@@ -0,0 +1,27 @@
+        padding:6px;
+        margin:0;
+        height: 27px;
+        font-family: 'Ubuntu', sans-serif;
```
Algumas regras tem sido colocados em todos os componentes, devem ser retiradas e colocadas como estilo global do pórprio StyledComponents

lucasnetwork replied:
corrigido

filiperochalopes commented on core/ui/src/components/Window/AsideBar/components/OptionBar/index.tsx:
```diff
@@ -0,0 +1,17 @@
+import React from 'react'
```
`.../components/Windown/components/AsideBar/components/OptionBar` O que seria essa OptionBar? Seria a barra de navegação? Se sim, poderia se chamar NavBar. Já que Nav já é um componente consagrado do HTML5 para isso.

lucasnetwork replied:
não seria uma navebar,seria mais um item dentro da asideBar

lucasnetwork replied:
pra facilitar mudei o nome pra ItemNavBar.

filiperochalopes commented on core/ui/src/components/Window/AsideBar/index.tsx:
```diff
@@ -0,0 +1,51 @@
+import React, { useEffect, useRef, useState } from 'react'
+import Container, { ButtonStyle, ButtonContainer } from './styles'
+import Button from '../../Button'
+/* global HTMLButtonElement */
```
O que são essas variáveis globais?

lucasnetwork replied:
quando eu usava alguns destas interfaces padrões da web, o eslint que estava retornando o erro "no-defined", e para corrigir isso sem ter que desabilitar essa regra no arquivo,tive que colocar esse comentario.

filiperochalopes commented on core/ui/src/components/Window/AsideBar/index.tsx:
```diff
@@ -0,0 +1,51 @@
+import useDimension from '../../../services/hooks/useDimension'
```
Novamente imports relativos

lucasnetwork replied:
corrigido

filiperochalopes commented on core/ui/src/components/Window/Header/components/HeaderButton/index.tsx:
```diff
@@ -0,0 +1,23 @@
+import Button from '../../../../Button'
```
Import relativo

lucasnetwork replied:
corrigido

filiperochalopes commented on core/ui/src/components/Window/Header/components/HeaderButton/index.tsx:
```diff
@@ -0,0 +1,23 @@
+import Button from '../../../../Button'
+import styled from 'styled-components'
+
+export default styled(Button)`
```
Estilos devem ser colocados em `styles.ts`

lucasnetwork replied:
corrigido

filiperochalopes commented on core/ui/src/components/Window/Header/index.tsx:
```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import HeaderButton from './components/HeaderButton'
+import Container from './styled'
```
Não podemos deixar o padrão, chamar uns de `styles` e outros de `styled`. Vamos manter `styles`

lucasnetwork replied:
corrigido

filiperochalopes commented on core/ui/src/components/Window/Header/index.tsx:
```diff
@@ -0,0 +1,48 @@
+  <Container>
+    <div className='button_header_container'>
+      <HeaderButton type="button" disabled={buttonDisabled} onClick={buttonClick}>
+        <svg
```
Os svgs devem ficar em `assets`

lucasnetwork replied:
corrigido

filiperochalopes commented on core/ui/src/components/Window/Header/styled.ts:
```diff
@@ -0,0 +1,27 @@
+import styled from 'styled-components'
```
`styles.ts`

lucasnetwork replied:
corrigido

filiperochalopes commented on core/ui/src/components/Window/Main/index.ts:
```diff
@@ -0,0 +1,7 @@
+import styled from 'styled-components'
```
`styles.ts`

lucasnetwork replied:
corrigido

filiperochalopes commented on core/ui/src/components/Window/Menu/components/Button/index.tsx:
```diff
@@ -0,0 +1,25 @@
+/* global HTMLButtonElement */
+
+const MenuButton:React.FC<ButtonHTMLAttributes<HTMLButtonElement>> = (props) => (
+    <Container type="button" {...props}>
```
Não entendi esse componente, vários elementos html vazios? a que se referem?

lucasnetwork replied:
basicamente,seria um icone que tem 9 pontos no menu inferior

filiperochalopes replied:
Ele forma um svg que poded ser exportado pelo próprio FIGMA, não faz sentido fazer um hardcode nele.

lucasnetwork replied:
ok então

lucasnetwork replied:
corrigido

filiperochalopes commented on core/ui/src/components/Window/Menu/components/ExplicationProgramIcon/index.tsx:
```diff
@@ -0,0 +1,24 @@
+        <div>
+            <div>
+            {icon}
+
```
Executar o eslint ou prettier para remover esses espaços e formatações desnecessárias e que dificultam a legibilidade do código.

lucasnetwork replied:
achei que o prettier tinha sido adicionado antes,por isso não verifiquei isso

filiperochalopes commented on core/ui/src/components/Window/Menu/style.ts:
```diff
@@ -0,0 +1,56 @@
+import styled, { css } from 'styled-components'
```
`styles.ts` cada arquivo está sendo nomeado de forma diferente `styles`, `styled`, `style`. Está bem bagunçado isso, o padrão é `styles`

lucasnetwork replied:
resolvido

filiperochalopes commented on core/ui/src/components/Window/Modal/index.tsx:
```diff
@@ -0,0 +1,21 @@
+import React from 'react'
```
O componente `Modal` não é um componente de Window, mas sim um componente a parte. Aliás a funcionalidade dele é diferente, deve ser como criação mesmo de um modal, o FIGMA inclusive mostra um modal sobre o outro. O modal deve ser algo:

```jsx
({title, content, returnButtonAction, confirmButtonAction}) => {
// Deve-se criar uma cortina dentro da div id="modal" em index.html com zIndex maior que as dedmais telas existentes e renderizar por cima da cortina um componente Modal que tem Header semelhante ao Window
}
```
Esse é só o rascunho, pode precisar de mais coisas.

filiperochalopes commented on core/ui/src/index.html:
```diff
@@ -3,6 +3,10 @@
+
   </head>
   <body>
     <div id="app"></div>
```
Adicionar o `<div id="modals"></div>` como descrito em comentários anteriores

lucasnetwork replied:
corrigido

filiperochalopes commented on core/ui/src/index.html:
```diff
@@ -3,6 +3,10 @@
     <title>PIN React Application</title>
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+<link href="https://fonts.googleapis.com/css2?family=Ubuntu:wght@300;400;700&display=swap" rel="stylesheet">
```
Alinhamento de texto, formatação.

lucasnetwork replied:
corrigido

filiperochalopes commented on core/ui/src/mocks/card.json:
```diff
@@ -0,0 +1,11 @@
+[
```
Deve ter conteúdo suficiente para exemplificar uma barra de rolagem.

lucasnetwork replied:
corrigido

filiperochalopes commented on core/ui/src/mocks/menu.json:
```diff
@@ -0,0 +1,17 @@
+    {
+        "id":"1",
+        "text":"Recepção",
+        "status":"Normal"
```
Por que os nomes maiúsculos nos status? Letras maiusculas são para classes e componentes. Você criou uma classe para status? Isso nada mais é do que uma definição de estilos, portanto deve manter-se minúscula.

lucasnetwork replied:
resolvido

filiperochalopes commented on core/ui/src/services/hooks/useDimension.ts:
```diff
@@ -0,0 +1,34 @@
+ * @returns
+ */
+
+// eslint-disable-next-line no-undef
```
Por que desabilitar essa função em vez de corrigir o erro?

lucasnetwork replied:
resolvido

filiperochalopes reviewed (commented):
Percebi que existe uma pasta temp com imagens png também que não foram excluídas. Devem ser excluídas, pois não fazem parte do projeto, já que os svg já estão sendo utilizados em `assets`

lucasnetwork committed:
feat(component): fix prettier and refactore Badge component

lucasnetwork committed:
fix(files): remove temp files

lucasnetwork committed:
fix(component): add badge optional in card component

lucasnetwork committed:
fix(svgs): rename name of svgs

lucasnetwork committed:
fix(component): change name of OptionBar to ItemNavBar component

lucasnetwork committed:
fix(hook service): fix error of eslint no-defined

lucasnetwork committed:
feat(component): create ContainerWindow

lucasnetwork committed:
refactor(component and page): fix style with prettier and rename file style name

lucasnetwork committed:
refactor(component): remove ButtonConfirm and add styles in Button component

move Styles of buttonConfirm to Button component that can to use like custom type of Button
component

lucasnetwork committed:
refactor(component): remove headerButton component and pass styles to Button component

lucasnetwork committed:
fix(component): create file tsx in button component

lucasnetwork committed:
feat(assets): add background image

lucasnetwork committed:
feat(webpack): add file-loader in webpack

lucasnetwork committed:
fix(component): fix padding of menu

lucasnetwork committed:
fix(component): fix z-index of header component

lucasnetwork committed:
refactor(interfaces): change names of interfaces,adding letter 'I' at the beginning of the word

lucasnetwork committed:
fix(component): move modals to div modals that is outside of root app

lucasnetwork committed:
style(html): styling code of index.html

lucasnetwork committed:
fix(component): fix type of ItemNavBar

lucasnetwork committed:
fix(component): remove tags in menu button to image svg

lucasnetwork committed:
fix(component): create index.tsx of content component

lucasnetwork committed:
fix(component): create index.tsx of content component

lucasnetwork committed:
refactor(component): change folders of form to components

lucasnetwork committed:
refactor(components): change folders in of windows to components

lucasnetwork committed:
feat(merge): merge branch with develop

lucasnetwork committed:
fix(paths): change paths to relative paths

lucasnetwork committed:
fix(component): fix scrollview of aside component

lucasnetwork committed:
feat(mock): add more items in menu.json

lucasnetwork committed:
feat(component): improvement modal component

lucasnetwork committed:
fix(component): fix select style

lucasnetwork committed:
feat(mocks): add new examples in card.json

lucasnetwork committed:
fix(component): add tag nav in asideBar

lucasnetwork committed:
feat(component): add button in ItemNavBar

lucasnetwork committed:
feat(component): remove box-shadow of button and remove prop useless in multiSelect component

lucasnetwork committed:
feat(dependence): add animate dependence

lucasnetwork committed:
feat(component): add initial animate

lucasnetwork committed:
feat(component): add Chart component

lucasnetwork committed:
fix(component): remove props and types of Card component

remove badges props of Card Button and change types of title and subtitle to allow ReactNode

lucasnetwork committed:
refactor(images): move svg image to assets icons

lucasnetwork committed:
feat(component): add animation of menu

lucasnetwork committed:
fix(dependence): remove animate dependence

lucasnetwork committed:
feat(styling): add global style

lucasnetwork committed:
feat(styling): add new global styles

lucasnetwork committed:
fix(component): fix distance of menu among program icons

lucasnetwork committed:
feat(mocks): add program mocks

lucasnetwork committed:
fix(mock and component): fix Program props and fix mock of programs

lucasnetwork committed:
feat(page): add confirm desability user modal

lucasnetwork committed:
feat(component): add default paragraph style

lucasnetwork committed:
feat(mocks): add mocks of chart

lucasnetwork committed:
fix(component): add button disabled in ItemNavBar when status is 'disabled'

lucasnetwork committed:
fix(page): align painel of page to end of page

lucasnetwork committed:
fix(component): fix scroll of searchInput

lucasnetwork committed:
fix(component): change Chart component name to HeaderChart and remove chart mock

lucasnetwork committed:
fix(component): fix styles of button when disabled

lucasnetwork committed:
fix(component): change card.json to card.tsx and change props of Card component

lucasnetwork committed:
fix(component): fix maxHeight of menuList of MultiSelect and Select components

lucasnetwork committed:
fix(page): fix bug of scroll and overflow of window

lucasnetwork committed:
feat(component): add hover and active styles in button component

lucasnetwork committed:
feat(eslint): add prettier in eslint

lucasnetwork committed:
feat(page): add apresentation page

lucasnetwork committed:
feat(component): add fixed attribute in HeaderChart

lucasnetwork committed:
feat(component): add contentLabel in card mock

lucasnetwork committed:
feat(component): add padding-bottom in ContainerWindow

lucasnetwork committed:
fix(component): fix hover style of button component

lucasnetwork committed:
fix(component): fix input label

fix active input when click in label, fix align-center and margin-right when label is left of input

lucasnetwork committed:
fix(page): fix menu mock and labelPosition of Input

lucasnetwork committed:
fix(page): fix scrollview

lucasnetwork committed:
fix(imports): fix imports to relative imports

lucasnetwork committed:
feat(component): add background component

lucasnetwork committed:
fix(component styles): remove styles inline of modal and move styles to components of modal

lucasnetwork committed:
feat(page): create Form and ContainerFormInput components styles of page

lucasnetwork committed:
feat(component): create OptionListContainer component

lucasnetwork committed:
feat(component): add ContainerContent and ContainerParagraphe component

lucasnetwork committed:
fix(component): remove use of containerProps in search component

lucasnetwork committed:
feat(dependence): add uuid dependence

add uuid dependence to create unique id in frontend to map keys

lucasnetwork committed:
feat(components,services): add ModalContext,modals components and change Modal component

add ModalContext to create Modals using addModal function, add modals into Modals components and
move react.createPortal of Modal component to ModalContext

lucasnetwork committed:
feat(context): making returnButtonAction and confirmButtonAction functions non-mandatory

lucasnetwork committed:
fix(component): fix hover and active styles button

lucasnetwork committed:
fix(folder): fix folder structure

lucasnetwork committed:
fix(component): fix button styles

lucasnetwork committed:
fix(mocks): change attributes names of card mock

lucasnetwork committed:
feat(page): add pages

lucasnetwork committed:
feat(routes): include routes

lucasnetwork committed:
feat(component): add hover and active style in button component green-confirm
